zebra-consensus/src/transaction.rs

```diff
@@ -153,175 +161,243 @@ where
+                // https://zips.z.cash/protocol/protocol.pdf#outputdesc
+                check::output_cv_epk_not_small_order(output)?;
+
+                // Consensus rule: The proof π_ZKOutput MUST be
+                // valid given a primary input formed from the other
+                // fields except C^enc and C^out.
+                //
+                // Queue the verification of the Groth16 output
+                // proof for each Output description while adding
+                // the resulting future to our collection of async
+                // checks that (at a minimum) must pass for the
+                // transaction to verify.
+                let output_rsp = output_verifier
+                    .ready_and()
+                    .await?
+                    .call(primitives::groth16::ItemWrapper::from(output).into());
+
+                async_checks.push(output_rsp.boxed());
+            }
+
+            let bvk = sapling_shielded_data.binding_verification_key();
+
+            // TODO: enable async verification and remove this block - #1939
+            {
+                let item: zebra_chain::primitives::redjubjub::batch::Item =
+                    (bvk, sapling_shielded_data.binding_sig, &shielded_sighash).into();
+                item.verify_single().unwrap_or_else(|binding_sig_error| {
+                    let binding_sig_error = binding_sig_error.to_string();
+                    tracing::warn!(%binding_sig_error, "ignoring");
+                    metrics::counter!("zebra.error.sapling.binding",
+                                                  1,
+                                                  "kind" => binding_sig_error);
+                });
+                // Ignore errors until binding signatures are fixed
+                //.map_err(|e| BoxError::from(Box::new(e)))?;
+            }
+
+            let _rsp = redjubjub_verifier
+                .ready_and()
+                .await?
+                .call((bvk, sapling_shielded_data.binding_sig, &shielded_sighash).into())
+                .boxed();
+
+            // TODO: stop ignoring binding signature errors - #1939
+            // async_checks.push(rsp);
+        }
+
+        // Finally, wait for all asynchronous checks to complete
+        // successfully, or fail verification if they error.
+        while let Some(check) = async_checks.next().await {
+            tracing::trace!(?check, remaining = async_checks.len());
+            check?;
+        }
+
+        Ok(tx.hash())
+    }
+
+    async fn verify_v5_transaction(
+        request: &Request,
+        network: Network,
+    ) -> Result<transaction::Hash, TransactionError> {
+        Self::verify_v5_transaction_network_upgrade(request.upgrade(network))?;
+
+        // TODO:
+        // - verify transparent pool (#1981)
+        // - verify sapling shielded pool (#1981)
+        // - verify orchard shielded pool (ZIP-224) (#2105)
+        // - ZIP-216 (#1798)
+        // - ZIP-244 (#1874)
```

Comment:
I'm not 100% sure if the linked issues are correct. I also omitted ZIP-225 because that looks like the V5 transaction encoding, so it did not seem relevant to this part of the code (and the issue I found related to that was complete).

Reply:
TODOs are useful, but we don't rely on them to track work, so I wouldn't worry about having the exact list of issues here.